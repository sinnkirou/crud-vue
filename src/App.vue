<template>
  <div id="app">
    <button
      id="demo-menu-lower-left"
      class="mdl-button mdl-js-button mdl-button--icon"
      type="button"
    >
      <i class="material-icons">more_vert</i>
    </button>
    <ul
      class="mdl-menu mdl-menu--bottom-left mdl-js-menu mdl-js-ripple-effect"
      for="demo-menu-lower-left"
    >
      <li class="mdl-menu__item">
        <router-link to="/">Home</router-link>
      </li>
      <li class="mdl-menu__item">
        <router-link to="/posts">Posts</router-link>
      </li>
    </ul>
    <router-view></router-view>
  </div>
</template>

<script>
import "./assets/stylesheets/material.min.js";
import uuid from "uuid/v1";

export default {
  name: "app",
  data: () => ({
    posts: [
      {
        id: "45745c60-7b1a-11e8-9c9c-2d42b21b1a3e",
        title: "My journey with Vue",
        message: "My journey with Vue"
      },
      {
        id: "45745c60-7b1a-11e8-9c9c-2d42b21b1a3f",
        title: "Blogging with Vue",
        message: "Blogging with Vue"
      },
      {
        id: "45745c60-7b1a-11e8-9c9c-2d42b21b1a3g",
        title: "Why Vue is so fun",
        message: "Why Vue is so fun"
      }
    ]
  }),
  methods: {
    addPost: function(formObj) {
      this.posts.push({
        id: uuid(),
        ...formObj
      });
    },
    deletePost: function(id) {
      this.posts = this.posts.filter(post => {
        return post.id != id;
      });
    },
    updatePost: function(formObj) {
      this.posts = this.posts.map(post =>
        post.id === formObj.id ? { ...formObj } : post
      );
    }
  },
  provide: function() {
    return {
      addPost: this.addPost,
      deletePost: this.deletePost,
      updatePost: this.updatePost
    };
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
@import "./assets/stylesheets/style.css";
@import "./assets/stylesheets/material.min.css";
@import "./assets/material-icons/material-icons.css";
</style>
