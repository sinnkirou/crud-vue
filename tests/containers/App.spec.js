import { shallowMount } from "@vue/test-utils";
import App from "@/App.vue";
import { localVue, store } from "../mock";

describe("App.vue", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(App, {
      localVue,
      store,
      stubs: ["router-link", "router-view"]
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  it("should render", () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  it("has the expected html structure", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
