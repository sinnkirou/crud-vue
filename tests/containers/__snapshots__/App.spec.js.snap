// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue has the expected html structure 1`] = `
<div
  id="app"
>
  <button
    class="mdl-button mdl-js-button mdl-button--icon"
    id="demo-menu-lower-left"
    type="button"
  >
    <i
      class="material-icons"
    >
      more_vert
    </i>
  </button>
   
  <ul
    class="mdl-menu mdl-menu--bottom-left mdl-js-menu mdl-js-ripple-effect"
    for="demo-menu-lower-left"
  >
    <li
      class="mdl-menu__item"
    >
      <router-link-stub
        to="/"
      >
        Home
      </router-link-stub>
    </li>
     
    <li
      class="mdl-menu__item"
    >
      <router-link-stub
        to="/posts"
      >
        Posts
      </router-link-stub>
    </li>
  </ul>
   
  <router-view-stub />
</div>
`;
